### PR TITLE
mu4e: properly display messages in the draft folder (with gnus)

### PR DESCRIPTION
**Pull request by Chris00**
This is necessary to display messages with UFT-8 chars when visiting the draft folder.  Here is an [example of such a message](https://github.com/djcb/mu/files/2824530/draft.txt).

### CODE REVIEW

**Comment by Chris00** [comment]
I added a commit that removes
```elisp
      (article-de-base64-unreadable)
      (article-de-quoted-unreadable)
```
because they actually *prevent* properly decoding base64 messages.  Here is  [an example](https://github.com/djcb/mu/files/2824600/decoding.txt)

File: mu4e/mu4e-view.el [hunk 1]
```diff
@@ -362,14 +362,17 @@ article-mode."
...
-      (mm-disable-multibyte)
       (insert-file-contents-literally path)
+      (message-narrow-to-headers-or-head)
+      (if (message-fetch-field "Content-Type" t)
+          (widen)
+        ;; For example, for messages in `mu4e-drafts-folder'
+        (erase-buffer)
+        (mm-enable-multibyte)
+        (insert-file-contents path))
       (setq
```
Comment (djcb) [inline mu4e/mu4e-view.el]: Can you explain what this block does and esp., why?

**Comment by djcb** [comment]
The change does seem to work with all "problem case" messages, great!

File: mu4e/mu4e-view.el (hunk 1 above)
Comment (Chris00) [reply mu4e/mu4e-view.el]: It inserts the message with multibytes turned on.  This enables the interpretation of UTF-8 byte sequences for draft messages — which do not have a charset.

Comment (djcb) [reply mu4e/mu4e-view.el]: Thanks, I see... can you add a comment tin the code that explains that?

Do we really need to re-load the file from disk (`insert-file-contents`)? Would be nice if we could avoid that.

Comment (Chris00) [reply mu4e/mu4e-view.el]: Thanks for prompting me to do a better job! ☺  I reworked the commit not to read the file twice.

**Comment by djcb** [comment]
Merged -- thanks for your efforts!

**Pull request merged**